packages/actor-http-cache/lib/ActorHttpCache.ts

```diff
@@ -0,0 +1,233 @@
+        newRequest,
+        new Response(cacheResult.body, cacheResult.init),
+        cacheResult.policy,
+        cacheStorage,
+      );
+    }
+    // The Cache Policy is not satisfied ---------------------------------------
+    // Invalidate the cache
+    await this.mediatorHttpInvalidate.mediate({ url: newRequest.url, context: action.context });
+    const revalidationHeaders = this.getRevalidationHeaders(newRequest, cacheResult.policy);
+    const response = await this.fetchDocument({
+      ...action,
+      init: {
+        ...action.init,
+        headers: revalidationHeaders,
+      },
+    });
+    const { policy, modified } = this.getRevalidatedPolicy(newRequest, response, cacheResult.policy);
+    const finalResponse = modified ? response : new Response(cacheResult.body, cacheResult.init);
+    return await this.teeAndCacheResponse(newRequest, finalResponse, policy, cacheStorage);
+  }
+
+  private async fetchDocument(action: IActionHttp): Promise<Response> {
+    const response = await this.mediatorHttp.mediate({
+      ...action,
+      context: action.context.set(KeysHttpCache.doNotCheckHttpCache, true),
+    });
+    return response;
+  }
+
+  private newCachePolicy(request: Request, response: Response): CachePolicy {
+    // Conversions to make headers compatible
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const responseWithHashHeaders = ActorHttpCache.responseToResponseWithHashHeaders(response);
+    return new CachePolicy(
+      requestWithHashHeaders,
+      responseWithHashHeaders,
+    );
+  }
+
+  private isPolicySatisfied(request: Request, policy: CachePolicy): boolean {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    return policy.satisfiesWithoutRevalidation(requestWithHashHeaders);
+  }
+
+  private getRevalidationHeaders(request: Request, policy: CachePolicy): Headers {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const hashRevalidationHeaders = policy.revalidationHeaders(requestWithHashHeaders);
+    return new Headers(<Record<string, string>> hashRevalidationHeaders);
+  }
+
+  private getRevalidatedPolicy(request: Request, response: Response, policy: CachePolicy): RevalidationPolicy {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const responseWithHashHeaders = ActorHttpCache.responseToResponseWithHashHeaders(response);
+    return policy.revalidatedPolicy(requestWithHashHeaders, responseWithHashHeaders);
+  }
+
+  private async teeAndCacheResponse(
+    request: Request,
+    response: Response,
+    cachePolicy: CachePolicy,
+    cacheStorage: IHttpCacheStorage,
+  ): Promise<Response> {
+    // Node-fetch does not support body.tee, while it is mandatory according to the fetch and readablestream api.
+    // If it doesn't exist, we monkey-patch it.
+    if (response.body && !response.body.tee) {
```

Comment:
Can we move this monkey-patch to after this line [here](https://github.com/comunica/comunica/blob/e89bc9f943c7358159eb720d1f638d092df2bef5/packages/actor-http-fetch/lib/ActorHttpFetch.ts#L110)?
Just so that we have all the fetch-related monkey-patching in a single place.